subiquity/common/types.py

```diff
@@ -747,6 +747,20 @@ class MirrorCheckResponse:
     output: str
 
 
+@attr.s(auto_attribs=True)
+class MirrorPost:
+    elected: Optional[str] = None
+    candidates: Optional[List[str]] = None
+    staged: Optional[str] = None
+
+
+@attr.s(auto_attribs=True)
+class MirrorGet:
+    elected: Optional[str]
+    candidates: List[str]
+    staged: Optional[str]
```

Comment:
I think the most obvious reason is that when doing a GET, having candidates=None is **not** acceptable, so having a single type with `candidates: Optional[List[str]] = None` is .. meh.

ideally IMO, the `MirrorPost` would be represented by a `TypedDict` with all keys optional ; rather than an `attr.s` with default `None` values everywhere ; but we don't support serialization with `TypedDict` classes (yet?).